model/task/task.go

```diff
@@ -3083,6 +3116,87 @@ func (t *Task) SetNumDependents() error {
 	}, update)
 }
 
+func AddExecTasksToDisplayTask(displayTaskId string, execTasks []string) error {
+	if len(execTasks) == 0 {
+		return nil
+	}
+
+	return UpdateOne(
+		bson.M{IdKey: displayTaskId},
+		bson.M{"$addToSet": bson.M{
+			ExecutionTasksKey: bson.M{"$each": execTasks},
+		}},
+	)
+}
+
+////////////////
+// Cedar Helpers
+////////////////
+
+// Get CedarTestResults fetches the task's test results from the Cedar service.
+// If the task does not have test results in Cedar, (nil, nil) is returned. If
+// the task is a display task, all of its execution tasks' test results are
+// returned.
+func (t *Task) GetCedarTestResults() ([]TestResult, error) {
+	ctx, cancel := evergreen.GetEnvironment().Context()
+	defer cancel()
+
+	if !t.hasCedarResults() {
+		return nil, nil
+	}
+
+	opts := apimodels.GetCedarTestResultsOptions{
+		BaseURL:   evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+		Execution: t.Execution,
+	}
+	if t.DisplayOnly {
+		opts.DisplayTaskID = t.Id
+	} else {
+		opts.TaskID = t.Id
+	}
+
+	cedarResults, err := apimodels.GetCedarTestResults(ctx, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting test results from cedar")
+	}
+
+	results := make([]TestResult, len(cedarResults))
+	for i, result := range cedarResults {
+		results[i] = ConvertCedarTestResult(result)
+	}
+
+	return results, nil
+}
+
+func (t *Task) hasCedarResults() bool {
+	if !t.DisplayOnly || t.HasCedarResults {
+		return t.HasCedarResults
+	}
+
+	// Older display tasks may incorrectly indicate that they do not have
+	// test results in Cedar. Since all or none of the execution tasks will
+	// have test results in Cedar, we only need to check one. In the case
+	// that there are results in Cedar, this will attempt to update the
+	// display task accordingly.
+	if len(t.ExecutionTasks) > 0 {
+		execTask, err := FindOneId(t.ExecutionTasks[0])
+		if err != nil || execTask == nil {
+			return false
+		}
+
+		// Attempt to update the display task's HasCedarResults field.
+		// We will not update the CedarResultsFailed field since we do
+		// want to iterate through all the execution tasks and it isn't
+		// really needed for display tasks.
+		// Since we do not want to fail here, we can ignore the error.
+		_ = t.SetHasCedarResults(execTask.HasCedarResults, false)
```

Comment:
if `execTask.HasCedarResults` is false is there a point to setting it on the display task?

Reply:
Not really, nothing will happen. I can make this an if statement if you want.

Reply:
that makes sense to me